src/uct/ib/dc/dc_mlx5.c

```diff
@@ -279,36 +279,34 @@ static UCS_F_ALWAYS_INLINE unsigned
 uct_dc_mlx5_iface_progress(void *arg, int flags)
 {
     uct_dc_mlx5_iface_t *iface = arg;
+    const int zip_rx           = iface->super.cq[UCT_IB_DIR_RX].zip;
+    const int zip_tx           = iface->super.cq[UCT_IB_DIR_TX].zip;
+    int zip_flag;
     unsigned count;
 
-    count = uct_rc_mlx5_iface_common_poll_rx(&iface->super, flags);
+    zip_flag = zip_rx ? UCT_IB_MLX5_POLL_FLAG_CQE_ZIP : 0;
+    count    = uct_rc_mlx5_iface_common_poll_rx(&iface->super, flags | zip_flag);
```

Comment:
seems it beats the purpose of resolving poll flag during compile time

Reply:
Yes, but I checked the performance and there is no degradation.

We can separate CQE_ZIP flag for TX and RX directions, but then we will have the exponential growth of `iface_progress` methods with different `poll_flags` configurations.

Reply:
> Yes, but I checked the performance and there is no degradation.

How did you check?

Reply:
Compared two OSU benchmarks execution results, one with the UCX built with this patch and other built without it.

Reply:
@ivankochin as discussed, please separate the funcs to "with" / "without" CQE zipping, aggregating all the CQE zipping "variants" in one place.